DEFEDIT-1133 Can delete open markdown file

ragnard-king opened the pull request:
Fixes https://github.com/defold/editor2-issues/issues/1110

### User facing changes

- You can now delete an open markdown file without error, and tab will
  close as expected.

### Technical changes

- Provide a `load-fn` when registering resource type for markdown, and
  load file as we do with other resources.

ragnard-king committed:
DEFEDIT-1133 Can delete open markdown file

* User facing changes

- You can now delete an open markdown file without error, and tab will
  close as expected.

* Technical changes

- Provide a `load-fn` when registering resource type for markdown, and
  load file as we do with other resources.

erikangelin-king reviewed (changes requested):
Delete works, but needs some polish :)

ragnard-king committed:
DEFEDIT-1133 Review fixes

- markdown property hidden
- do markdown to html conversion in html output instead of `load-fn`
